File: python/Array/sum of digits.py
#Find the sum of the digits of the number using a recursive function.


def _findSum(list, N):
    if len(list) == 1:
        return list[0]
    else:

        return list[0] + _findSum(list[1:], N)

n = input('enter the number without any spaces (only one number with no limit of digits): ')
list = [int(d) for d in str(n)]

N = len(list)
ans = _findSum(list, N)
print('the sum of digits are: ', ans)

File: Java_Programs/Array/sumOfDigits.java
import java.util.*;
import java.util.Scanner;
public class sumOfDigits {
	static int sum(int n) {
		if (n==0) 
			return 0;
			return(n%10 +sum(n/10));
		}
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		System.out.print("enter any number: ");
		int a=sc.nextInt();
		int result=sum(a);
		System.out.println("sum of digits in "+a+" is: "+result);
	}
}
